cpp/src/arrow/compute/kernel.cc

```diff
@@ -411,21 +409,23 @@ bool InputType::Matches(const DataType& type) const {
       return type_->Equals(type);
     case InputType::USE_TYPE_MATCHER:
       return type_matcher_->Matches(type);
-    default:
-      // ANY_TYPE
-      return true;
+    case InputType::ANY_TYPE:
+      break;
```

Comment:
It's not a strong opinion but I like using `return true` here for consistency of other `case`s in this `switch`:

```suggestion
      return true;
```

Reply:
OK. I'm changing this to `true` and `false` after the `switch`.